File: OraculumApi/Models/FrontOffice/Feedback.cs

namespace OraculumApi.Models.FrontOffice
{
    public partial class Feedback : WeaviateEntity, IEntity<FeedbackDTO>
    {
        public string? text;
        public string? rating;
        public string? sibyllaId;
        public string? chatId;
        public string? messageId;
        public string? questionMessageId;
        public string? factId;
        public DateTime timestamp;

        public FeedbackDTO toDTO()
        {
            return new FeedbackDTO()
            {
                Id = this.id,
                Text = this.text,
                Rating = this.rating,
                SibyllaId = this.sibyllaId,
                ChatId = this.chatId,
                MessageId = this.messageId,
                QuestionMessageId = this.questionMessageId,
                FactId = this.factId,
                Timestamp = this.timestamp
            };
        }
    }
}

File: OraculumFE/src/app/interfaces/feedback.ts
// interfaces/feedback.ts

export interface FeedbackBase {
  text?: string;
  rating: string;
}

export interface FeedbackWithMessageId extends FeedbackBase {
  messageId: string;
}

export interface FeedbackWithFactId extends FeedbackBase {
  factId: string;
}

export type FeedbackOptions = {
  chatId?: string;
  factId?: string;
} & (FeedbackBase | FeedbackWithMessageId | FeedbackWithFactId);
